### ISSUE TITLE
[New Rule] Getters SHOULD NOT change the state of an object

### ISSUE THREAD

**Issue by lenaorobei** [magento/magento-coding-standard#26]
### Rule
Getters SHOULD NOT change the state of an object.

### Reason
Source: [Magento Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html).

### Implementation
Check if method name starts with `get` and find class member assignment operation inside this method.
```
<severity>8</severity>
<type>warning</type>
```

**Comment by larsroettig** [comment]
Hi @paliarush and @lenaorobei,
I think we get too many false positives it is very hard to Implementing this with tokens maybe with phpstan it is possible to implement this check.
